gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagActionStore.java

```diff
@@ -28,6 +28,12 @@
 import org.apache.gobblin.service.modules.flowgraph.DagNodeId;
 
 
+/**
+ * This interface defines the generic APIs for storing and retrieval of {@link DagAction}s in GaaS. This store maintains
+ * state about the new actions to be performed on a flow or job, which can be initiated by either client requests or
+ * the GaaS system itself to progress a flow execution and enforce system constraints. The flow management layer
+ * retrieves these new requests and executes them.
```

Comment:
nit: no need to document that an interface is an interface (or a class is a class) or that interfaces/classes have APIs - viz. `This interface defines the generic APIs...` .  

try e.g.:
```
GaaS store for to-be-performed {@link DagAction}s on a flow or job.
These are created and stored by client request or generated within GaaS.
The Flow Management layer retrieves and executes {@link DagAction}s to progress a flow's execution or enforce execution deadlines.
```